src/Avalonia.Controls.ColorPicker/ColorPalettes/FlatColorPalette.cs

```diff
@@ -289,8 +289,10 @@ public enum FlatColor : uint
         };
 
         // See: https://htmlcolorcodes.com/assets/downloads/flat-design-colors/flat-design-color-chart.png
+#pragma warning disable CA2211 // Non-constant fields should not be visible
         protected static Color[,]? _colorChart = null;
-        protected static object _colorChartMutex = new object();
+#pragma warning restore CA2211 // Non-constant fields should not be visible
+        protected static readonly object _colorChartMutex = new();
```

Comment:
My comments here apply to all 4 affect color palettes.

1. Firstly I'm not a fan of rule CA2211 for exactly this reason. It doesn't allow for protected fields which absolutely have a usage and don't violate the non-public principle. The rule needs to be fixed...
2. With that said, the table itself was made protected before the final design was complete. There was a thought that it would be needed in derived types. However, with the colors now held in an enum (compiled in for performance) and then the inability to share color charts between half/full palettes, I think this design is obsolete. My suggestion is to make both of these fields `private`.
3. I don't care about the `new()` syntax usage. This is still largely developer preference so I probably wouldn't have changed it but it doesn't matter. Avalonia seems to prefer the shortest possible code.

```
private static Color[,]? _colorChart = null;
private static object _colorChartMutex = new();
```